device_backends/uio/src/UioAccess.cc

```diff
@@ -9,14 +9,16 @@
 #include <cerrno>
 #include <fcntl.h>
 #include <fstream>
+#include <iostream>
 #include <limits>
 #include <poll.h>
 
 namespace ChimeraTK {
 
   UioAccess::UioAccess(const std::string& deviceFilePath) : _deviceFilePath(deviceFilePath.c_str()) {
     std::string fileName = _deviceFilePath.filename().string();
-    _deviceKernelBase = (void*)readUint64HexFromFile("/sys/class/uio/" + fileName + "/maps/map0/addr");
+    //NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+    _deviceKernelBase = reinterpret_cast<void*>(readUint64HexFromFile("/sys/class/uio/" + fileName + "/maps/map0/addr"));
     _deviceMemSize = readUint64HexFromFile("/sys/class/uio/" + fileName + "/maps/map0/size");
     _lastInterruptCount = readUint32FromFile("/sys/class/uio/" + fileName + "/event");
 
```

Comment:
I know this is not your code, but in line 28 a `runtime_error` is thrown, which should not be done in the constructor of the backend. When does this code run in the end? And why isn't the DeviceFile class used here?

Reply:
Yeah that make stuff easier I suppose. will change that. That would also move the runtime error into open